doc/custom_parallel_backend.rst

```diff
@@ -0,0 +1,291 @@
+.. _custom_parallel_backend:
+
+==================================
+Parallel backend customization API
+==================================
+
+User can provide their own implementation of a parallel processing backend in
+addition to the ``'loky'``, ``'threading'``, ``'multiprocessing'`` backends
+provided by default.
+
+A new backend is defined by providing a backend factory, which is then
+registered with the :func:`joblib.register_parallel_backend` function by
+passing a name and the factory.
+
+For instance, to register a custom backend named ``'custom'`` with a factory ``MyCustomBackend``, you can call:
+
+.. code-block:: python
+
+    # Register the backend so it can be used with parallel_config
+    register_parallel_backend('custom', MyCustomBackend)
+
+This backend can then be used within the ``parallel_config`` context manager,
+as:
+
+.. code-block:: python
+
+    from joblib import Parallel, delayed, parallel_config
+
+    with parallel_config("custom"):
+        res = Parallel(2)(delayed(id)(i) for i in range(10))
+
+
+Minimal backend factory specification
+======================================
+
+The backend factory can be any callable that returns an instance of
+``ParallelBackendBase``. The complete specification of the backend factory is
+referenced in the `default backends source code`_.
+Here, we provide minimal details to implement a custom backend factory and
+the next section describes various possible customizations.
+
+The key methods for the backend factory are:
+
+- ``configure``: this method is called when creating a particular instance of
+  the ``Parallel`` object. It can allocate the resources needed for the
+  parallel computation and return the effective number of jobs that can be run
+  in parallel.
+- ``effective_n_jobs``: This method is called to determine the number of jobs
+  that can be run in parallel. It can be used to limit the number of jobs that
+  can be run in parallel.
```

Comment:
```suggestion
- ``effective_n_jobs``: This method is called to determine the number of jobs
  that can be run in parallel. It can be used to limit the number of jobs depending
  on the specified ``n_jobs`` and the system it runs on.
```